src/lib/components/layout/Grid/helpers.js

```diff
@@ -0,0 +1,19 @@
+export const generateResponsiveCustomProperties = (prop, infix) => {
+  if (typeof prop === 'undefined') {
+    return null;
+  }
+
+  const customProperties = {};
```

Comment:
I would replace it with:

```js
  if (typeof prop !== 'object') {
    return { [`--rui-local-${infix}-xs`]: prop };
  }

  return Object.keys(prop).reduce((acc, breakpoint) => ({
    ...acc,
    [`--rui-local-${infix}-${breakpoint}`]: prop[breakpoint],
  }), {});
```
The return early pattern makes it easier to follow as I can exit the function, well, early as the logical branches do not reconnect.